File: python/src/util/linkedlist.py
class LinkedList:

    class ListNode:

        def __init__(self, value):
            self.value = value
            self.prev = None
            self.next = None

    class LinkedListIterator:

        def __init__(self, head):
            self.current = head

        def __iter__(self):
            return self

        def __next__(self):
            if self.current is None:
                raise StopIteration

            value = self.current.value
            self.current = self.current.next
            return value

    def __init__(self):
        self.head = None
        self.tail = None
        self.list_size = 0

    def __iter__(self):
        return self.LinkedListIterator(self.head)

    def insert_front(self, value):
        pass

    def insert_end(self, value):
        pass

    def remove_front(self):
        return None

    def remove_end(self):
        return None

    def remove(self, value):
        pass

    def front(self):
        return None

    def back(self):
        return None

    def contains(self, value):
        return False

    def size(self):
        return 0

    def is_empty(self):
        return False

File: java/src/edu/berkeley/cs/util/LinkedList.java
package edu.berkeley.cs.util;

import java.util.Iterator;

public class LinkedList<T> implements Iterable<T> {
  static class ListNode<T> {
    ListNode<T> prev;
    ListNode<T> next;
    T value;

    ListNode(T value) {
      this.value = value;
    }
  }

  protected ListNode<T> head;
  protected ListNode<T> tail;
  private int size = 0;

  @Override
  public Iterator<T> iterator() {
    return new Iterator<T>() {
      ListNode<T> current = head;

      @Override
      public boolean hasNext() {
        return current != null;
      }

      @Override
      public ListNode<T> next() {
        ListNode<T> prev = current;
        current = current.next;
        return prev;
      }
    };
  }

  public void insertFront(T value) {}

  public void insertEnd(T value) {}

  public T removeFront() {
    return null;
  }

  public T removeEnd() {
    return null;
  }

  public boolean remove(T value) {
    return false;
  }

  public T front() {
    return null;
  }

  public T back() {
    return null;
  }

  public boolean contains(T value) {
    return false;
  }

  public int size() {
    return 0;
  }

  public boolean isEmpty() {
    return false;
  }
}
